plonky2/src/plonk/circuit_builder.rs

```diff
@@ -271,6 +271,10 @@ impl<F: RichField + Extendable<D>, const D: usize> CircuitBuilder<F, D> {
         );
     }
 
+    pub fn add_gate_to_gate_set(&mut self, gate: GateRef<F, D>) {
```

Comment:
Is this public because users will need to call it for cyclic recursion?

Reply:
Hm I don't think users would call this directly but I thought it could potentially be useful to people using `plonky2` as a library.